src/main/connection_listener/exchange_msg.rs

```diff
@@ -372,7 +373,8 @@ impl<UID: Uid> ExchangeMsg<UID> {
             }
         }
 
-        match self.socket.write(poll, self.token, msg) {
+        // TODO(povilas): use authed encryption
```

Comment:
This should always be authenticated encryption as it's _responding_ to someone meaning there was a prior read where that someone must have given their keys by now.

So all of [these reads](https://github.com/maidsafe/crust/pull/1084/files#diff-5f3070043b7a5b205deab9f8f768a619R130) should also give the public-key to this state.